pkg/ccl/backupccl/restore_data_processor.go

```diff
@@ -417,6 +430,7 @@ func (rd *restoreDataProcessor) processRestoreSpanEntry(
 		false, /* splitFilledRanges */
 		rd.flowCtx.Cfg.BackupMonitor.MakeBoundAccount(),
 		rd.flowCtx.Cfg.BulkSenderLimiter,
+		rd.spec.ValidateOnly,
```

Comment:
yeah i waffled about this. Pushing the logic into the batcher allows use to use the current checkpointing and update job progress machinery (note the `batcher.getSummary()` call at the bottom of `processRestoreSpanEntry`.

To skip some steps in the batcher, i could add a custom vaidationOnly summary builder in the batcher.

Reply:
alternatively, we could make an interface describing what we need of the batcher here, and then swap a mock in if we're in verification (similar to the splitter?)

Reply:
Aha! Great idea. Done.

Reply:
Not _quite_ what I meant: can we move the interface and noop version to backupccl/restore_data_processor.go and use it instead of calling MakeSSTBatcher in the validation case?

Reply:
aha. bc other users of the regular batcher don't need to know about the noop batcher? sounds good. done.